File: database8.py
# delete query
import mariadb
import sys
try:
           conn=mariadb.connect(
                      user="root",
                      password="maria",
                      host="127.0.0.1",
                      port =3308,
                      database="dhiraj"
                      )
except mariadb.Error as e:
           print("Error to connect",e)
           sys.exit(1)

print("Database connected successfully :)")
# get cursor
cur=conn.cursor()

try:
    cur.execute("delete from student where id=102")
    #delete the record where id is equal to 102
except mariadb.Error as e:
    print("Error",e)

conn.commit() # save the changes

print() # for new line

print("Data Deleted successfully :)\n")

print("After deletion the records \n")

try:
    cur.execute("select * from student")
    # retrieve all the records

except mariadb.Error as e:
    print("Error",e)

print() # for new line

for i,j,k in cur:
    print("id",i,"name",j,"age",k)

# closing the connection

conn.close()

File: database8.js
//select particular columns
var  mysql=require("mysql")

var conn= mysql.createConnection({host:"localhost",user:"root",password:"rootpassword",database:"node"})

conn.connect(function(err){
    if(err ) throw err;
    console.log("connected :)");

    //only returns only id, name from  the table

    conn.query("select id ,name from nodetable",function(err,result,fields){
        if(err) throw err;
        console.log(result);//returns whole records

        console.log("The name of the third record is :");

        console.log(result[2].name); //returns name of the third record

        console.log("The details of the fields :");

        console.log(fields) ; //prints the details of fields

        //returns the name of second field

        console.log("The name of the second field is: ");
        console.log(fields[1].name); //returns name 

        console.log("Thanks for using Node.js :)");

    })
});
